plugins/Goals/API.php

```diff
@@ -402,6 +403,14 @@ public function getMetrics($idSite, $period, $date, $segment = false, $idGoal =
         $allMetrics = Goals::getGoalColumns($idGoal);
         $requestedColumns = Piwik::getArrayFromApiParameter($columns);
 
+        $shouldAddAverageOrderRevenue = (in_array('avg_order_revenue', $requestedColumns) || empty($requestedColumns)) && $isEcommerceGoal;
+
+        if ($shouldAddAverageOrderRevenue && !empty($requestedColumns)) {
+            $avgOrder = new AverageOrderRevenue();
+            $metricsToAdd = $avgOrder->getDependentMetrics();
+            $requestedColumns = array_unique(array_merge($requestedColumns, $metricsToAdd));
+        }
+
```

Comment:
This logic should be handled by the Report::getMetricsRequiredForReport() call below, is there a reason it isn't?